src/sentry/api/event_search.py

```diff
@@ -662,3 +662,150 @@ def get_snuba_query_args(query=None, params=None):
+        raise InvalidSearchQuery("Unknown aggregate function '%s'" % field)
+
+    function_data = VALID_AGGREGATES[function_name]
+    column = match.group('column')
+    if column not in function_data['fields'] and function_data['fields'] != '*':
+        raise InvalidSearchQuery(
+            "Invalid column '%s' in aggregate function '%s'" % (column, function_name))
+
+
+def validate_orderby(orderby, fields):
+    orderby = orderby if isinstance(orderby, (list, tuple)) else [orderby]
+    for column in orderby:
+        column = column.lstrip('-')
+        if column not in fields:
+            raise InvalidSearchQuery('Cannot order by an field that is not selected.')
+
+
+def resolve_field_list(fields, snuba_args):
+    """
+    Expand a list of fields based on aliases and aggregate functions.
+
+    Returns a dist of aggregations, selected_columns, and
+    groupby that can be merged into the result of get_snuba_query_args()
+    to build a more complete snuba query based on event search conventions.
+    """
+    # If project.name is requested, get the project.id from Snuba so we
+    # can use this to look up the name in Sentry
+    if 'project.name' in fields:
+        fields.remove('project.name')
+        if 'project.id' not in fields:
+            fields.append('project.id')
+
+    aggregations = []
+    groupby = []
+    columns = []
+    for field in fields:
+        if field in FIELD_ALIASES:
+            special_field = deepcopy(FIELD_ALIASES[field])
+            columns.extend(special_field.get('fields', []))
+            aggregations.extend(special_field.get('aggregations', []))
+            continue
+
+        # Basic fields don't require additional validation. They could be tag
+        # names which we have no way of validating at this point.
+        match = AGGREGATE_PATTERN.search(field)
+        if not match:
+            columns.append(field)
+            continue
+
+        validate_aggregate(field, match)
+        aggregations.append([
+            VALID_AGGREGATES[match.group('function')]['snuba_name'],
+            match.group('column'),
+            u'{}_{}'.format(match.group('function'), match.group('column'))
+        ])
+
+    rollup = snuba_args.get('rollup')
+    if not rollup:
+        # Ensure fields we require to build a functioning interface
+        # are present. We don't add fields when using a rollup as the additional fields
+        # would be aggregated away.
+        if 'project.id' not in columns:
+            columns.append('project.id')
+        if not aggregations and 'id' not in columns:
+            columns.append('id')
+        if aggregations and 'latest_event' not in fields:
+            columns.extend(deepcopy(FIELD_ALIASES['latest_event']['fields']))
+
+    basic_fields = [col for col in columns if isinstance(col, six.string_types)]
+    if rollup and basic_fields:
+        raise InvalidSearchQuery('You cannot combine rollup with non-aggregate fields')
```

Comment:
Isn't that ok since they get added to groupby clause?

Reply:
Maybe, my understanding of rollup is that it cuts the data up into timeslices. Getting a non-aggregate column in that kind of result set is not helpful as it breaks the rollup slices.

Reply:
Why won't it be useful? I could see something like Errors by version rolled up by week being pretty useful.

Reply:
Fair point. I'll take this out.

Reply:
On second thought, this scenario is an error if there are no aggregates being included. However, basic fields + aggregates should be valid.

Reply:
Ok